src/routes.js

```diff
@@ -111,7 +111,7 @@ const transactionsHistory = (dbApi: DbApi, { logger, apiConfig }: ServerConfig)
   validateDatetimeReq(req.body)
   logger.debug('[transactionsHistory] request is valid')
   const result = await dbApi.transactionsHistoryForAddresses(
-    apiConfig.txHistoryResponseLimit,
+    req.body.txLimit || apiConfig.txHistoryResponseLimit,
```

Comment:
If I understand right, a client may force with this a very big amount of transactions to be returned - there are indeed addresses with very long history in the blockchain which may be used to DDoS the server with few of those requests